### PR TITLE
Added notebook similar_products_recommender_using_matching_engine

### PR DESCRIPTION
**Pull request by sudarshan-SpringML**
Added notebook **notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb**. 
This notebook describes how to use the  Vertex AI Matching Engine's Approximate Nearest Neighbor (ANN) service to create a similar products recommender

- [X] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [X] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [X] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [X] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [X] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

### CODE REVIEW

File: notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb [hunk 1]
```diff
@@ -0,0 +1,1683 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <h1>Text-Based Product Embeddings -&gt; lowercase</h1><p> </p><p>Add missing Colab link</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Service -&gt; service</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p><p>Vertex SDK -&gt; Vertex AI SDK</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Explore the data</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Combine the next 2 steps,</p><p>Add heding</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>This says cell type is "raw cell" not code cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>combine 2 code cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>remove (Redundant): The model is available to us via the TFHub.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heding</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading. Combine the 2 code cells</p><p>print(entence_embeddings[0])</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p><p>GCS -&gt; Cloud Storage</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='62'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Create the indexes</p><p> </p><p>A ANN  -&gt; An ANN</p><p> </p><h3>Create ANN Index (for Production Usage) -&gt; Create an ANN index for production usage</h3>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='64'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p><p> </p><p>Please read the documentation -&gt; Read the documentation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='66'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>combine 2 code cells</p><p> </p><p>print(INDEX_RESOURCE_NAME)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='67'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='69'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Create brute for index for ground truths</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='71'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Combine 2 code cells</p><p> </p><p>print(INDEX_BRUTE_FORCE_RESOURCE_NAME)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='72'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add text cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='74'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>print(VPC_NETWORK_FULL)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='76'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='77'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>combine code cells</p><p> </p><p>print(NDEX_ENDPOINT_NAME)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='78'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>combine 2 text cells</p><p> </p><p>Deploy Indexes -&gt; Deploy the Indexes</p><p> </p><p>Deploy ANN Index -&gt; Deploy the ANN Index</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='80'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Deploy ANN -&gt; Deploy the ANN</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='83'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Deploy -&gt; Deploy the</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='85'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>I think this text needs to be moved to the above text cell, and then combine 2 code cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='87'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <h2>Online Queries -&gt; lowercase</h2>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='89'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add headibng</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='91'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p><p> </p><p>2d -&gt; 2-dimensional</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='93'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Test query -&gt; Test the query. (Increase the font size)</p><p> </p><p>The split between 1. and 2. does not make sense</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='95'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>print(response_for_search_query)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='97'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='98'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>remove text cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='100'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>combine with above code cell</p><p>print(results_id_list)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='101'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='102'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <h4>Recommend almost similar products  -&gt; Recommend the similar products (INCREASE FONT SIZE)</h4>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='104'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p><p> </p><p>print(words_present)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='106'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p><p>Calculate length -&gt; Calculate the length </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='108'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Increase font of heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='110'/>

Comment (andrewferlitsch) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='112'/>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>DONE</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>changed to markdown</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>updated</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <h4>Please find that this section is to recommend almost similar products. Recommend the similar products(this heading points to above section)</h4><p> </p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>done</p>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>Please find that notebook cannot be run in colab, as this notebook has to be executed in a notebook which is created in a vpc network created in this notebook. It has dependency on vpc network created in this notebook.</p>

File: notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb [hunk 2]
```diff
@@ -0,0 +1,1569 @@
...
```
Comment (ivanmkc) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <p>See boilerplate reduction instructions and use the new notebook_template.ipynb</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (ivanmkc) [inline notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.            f.write('{"id":"' + str(index) + '",')
</pre><p>Use Python dictionary and json.dump</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1212/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

Comment (sudarshan-SpringML) [reply notebooks/official/matching_engine/similar_products_recommender_using_matching_engine.ipynb:1]: done

**Review by andrewferlitsch — APPROVED** [review]

**Comment by andrewferlitsch** [comment]
Step #4: File /workspace/workspace/env/lib/python3.9/site-packages/pkg_resources/__init__.py:1611, in DefaultProvider._get(self, path)
Step #5: Step #4:    1610 def _get(self, path):
Step #5: Step #4: -> 1611     with open(path, 'rb') as stream:
Step #5: Step #4:    1612         return stream.read()
Step #5: Step #4: 
Step #5: Step #4: FileNotFoundError: [Errno 2] No such file or directory: '/workspace/workspace/env/lib/python3.9/site-packages/google_api_core-2.11.0.dist-info/METADATA'
Step #5: Step #4:

**Comment by ivanmkc** [comment]
Since Sudarshan won't work on this notebook anymore, we'll need to branch off of his branch and create a new PR.

**Comment by andrewferlitsch** [comment]
@sudarshan-SpringML    please fetch and rebase on main to get fixes to the vertex-ai-notebook-execution-test check.

**Comment by sudarshan-SpringML** [comment]
> @sudarshan-SpringML please fetch and rebase on main to get fixes to the vertex-ai-notebook-execution-test check.

done. Rebase on main done

**Comment by andrewferlitsch** [comment]
TimeoutError: Operation did not complete within the designated timeout of 900 seconds.

**Comment by ivanmkc** [comment]
@andrewferlitsch I'm closing this PR as we already have a text embedding example. I wanted one that did product embeddings based on user interest and other modalities outside of text. Seems like this one just uses the product name.

**Pull request closed without merging**